tests/suit/framework/ovirt.go

```diff
@@ -0,0 +1,257 @@
+package framework
+
+import (
+	"fmt"
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1/ref"
+	"github.com/konveyor/forklift-controller/pkg/controller/provider/container/ovirt"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	corev1 "k8s.io/api/core/v1"
+	"os"
+)
+
+const DefaultStorageClass = "standard"
+
+func (r *OvirtClient) setupClient() (err error) {
+	r.CustomEnv = false
+	if env := os.Getenv("OVIRT_CUSTOM_ENV"); env == "true" {
+		r.CustomEnv = true
+	}
+
+	r.storageClass = DefaultStorageClass
+	if sc := os.Getenv("STORAGE_CLASS"); sc != "" {
+		r.storageClass = sc
+	}
+
+	mandEnvVars := []string{"OVIRT_USERNAME", "OVIRT_PASSWORD", "OVIRT_URL", "OVIRT_CACERT", "OVIRT_VM_ID"}
+	for _, field := range mandEnvVars {
+		if env := os.Getenv(field); env == "" {
+			return fmt.Errorf("%s is not set", field)
+		}
+	}
+
+	cacertFile := os.Getenv("OVIRT_CACERT")
+	fileinput, err := os.ReadFile(cacertFile)
+	if err != nil {
+		return fmt.Errorf("could not read %s", cacertFile)
+	}
+
+	r.Username = os.Getenv("OVIRT_USERNAME")
+	r.Password = os.Getenv("OVIRT_PASSWORD")
+	r.OvirtURL = os.Getenv("OVIRT_URL")
+	r.testVMId = os.Getenv("OVIRT_VM_ID")
+	r.Cacert = fileinput
+
+	return
+}
+
+// LoadSourceDetails - Load Source VM details from oVirt
+func (r *OvirtClient) LoadSourceDetails() (err error) {
+
+	// default test values
+	sdomains := []string{"95ef6fee-5773-46a2-9340-a636958a96b8"}
+	nics := []string{"6b6b7239-5ea1-4f08-a76e-be150ab8eb89"}
+
+	// if a real custom ovirt environment is used.
+	if r.CustomEnv {
+		// connect to ovirt instance
+		err = r.Connect()
+
+		// get storage domain from the test VM
+		sdomains, err = r.getSDFromVM(ref.Ref{ID: r.testVMId})
+		if err != nil {
+			return fmt.Errorf("error getting storage domains from VM - %v", err)
+		}
+
+		// get network interface from the test VM
+		nics, err = r.getNicsFromVM(ref.Ref{ID: r.testVMId})
+		if err != nil {
+			return fmt.Errorf("error getting network interfaces from VM - %v", err)
+		}
+
+	}
+	r.vmData.sdPairs = sdomains
+	r.vmData.nicPairs = nics
+
+	return
+}
+
+// ConnectUsingSecret - Connect to the oVirt API using secret.
```

Comment:
need to update the comment

Reply:
done, I've  simplified it so we wont need the secret in this point.